### PR TITLE
improve ci readability and smoke test flakiness

### PR DESCRIPTION
**Pull request by davidiw**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,14 +44,19 @@ jobs:
     steps:
       - dev-setup
       - run: docker run --detach -p 5432:5432 cimg/postgres:14.2
-      - run: INDEXER_DATABASE_URL="postgresql://postgres@localhost/postgres" RUST_BACKTRACE=full cargo nextest --nextest-profile ci --partition hash:1/1 --package smoke-test --retries 3
+      - run: echo "export RUST_BACKTRACE=full" >> $BASH_ENV
+      - run: echo "export INDEXER_DATABASE_URL=postgresql://postgres@localhost/postgres" >> $BASH_ENV
+      # --test-threads is intentionally set to reduce resource contention in ci jobs. Increasing this, increases job failures and retries
+      - run: cargo nextest --nextest-profile ci --partition hash:1/1 --package smoke-test --test-threads 6 --retries 3
```
Comment (zekun000) [inline .circleci/config.yml:50]: 6 is still a lot, I remember we were at 3 before?

Comment (davidiw) [reply .circleci/config.yml:50]: 5

**Review by gregnazario — APPROVED** [review]

**Comment by davidiw** [comment]
This is fascinating, somehow main is broken lol ... the api tests need to be regenerated.

**Comment by zekun000** [comment]
I think it's broken because the backtrace env is changed differently

**Review by CapCap — APPROVED** [review]
CI warrior,
Bravely protecting the merge,
Fixes flakey tests

File: .circleci/config.yml (hunk 1 above)
Comment (JoshLind) [reply .circleci/config.yml:50]: Why not 5 again? :)

Comment (davidiw) [reply .circleci/config.yml:50]: I like to live dangerously

**Review by areshand — APPROVED** [review]

**Pull request merged**